### PR TITLE
Revert player iterator

### PR DESCRIPTION
**Pull request by TimGoll**
partially reverts #1453

Not sure if this is the best solution or if we should still keep some player.Iterator() calls. Feel free to discuss.

See [this discussion on the GMod Discord](https://discord.com/channels/565105920414318602/567617926991970306/1224041147962228746) for more context on this issue.

### CODE REVIEW

**Comment by EntranceJew** [comment]
Not all of these cases are trying to modify the table, wouldn't make sense to revert them all across the board

File: gamemodes/terrortown/gamemode/server/sv_weaponry.lua [hunk 1]
```diff
@@ -13,7 +13,7 @@ local IsValid = IsValid
 local table = table
 local timer = timer
 local hook = hook
-local playerIterator = player.Iterator
+local playerGetAll = player.GetAll
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/server/sv_weaponry.lua:16]: Remove this one
```suggestion
```

File: gamemodes/terrortown/gamemode/shared/sh_player_ext.lua [hunk 2]
```diff
@@ -8,7 +8,7 @@ local table = table
 local IsValid = IsValid
 local hook = hook
 local math = math
-local playerIterator = player.Iterator
+local playerGetAll = player.GetAll
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/shared/sh_player_ext.lua:11]:
```suggestion
```

File: lua/ttt2/extensions/util.lua [hunk 3]
```diff
@@ -89,7 +89,7 @@ end
 -- @return table
 -- @realm shared
 function util.GetFilteredPlayers(filterFn)
-    local plys = select(2, playerIterator())
+    local plys = player.GetAll()
```
Comment (saibotk) [inline lua/ttt2/extensions/util.lua:92]:
```suggestion
    local plys = playerGetAll()
```

File: lua/ttt2/extensions/util.lua [hunk 4]
```diff
@@ -111,7 +111,7 @@ end
 -- @return table
 -- @realm shared
 function util.GetAlivePlayers()
-    local plys = select(2, playerIterator())
+    local plys = player.GetAll()
```
Comment (saibotk) [inline lua/ttt2/extensions/util.lua:114]:
```suggestion
    local plys = playerGetAll()
```

File: lua/ttt2/libraries/entspawnscript.lua [hunk 5]
```diff
@@ -491,7 +491,7 @@ if SERVER then
     -- @param table plys A table of players that should be updated
     -- @realm server
     function entspawnscript.UpdateSettingsOnClients(plys)
-        plys = plys or select(2, playerIterator())
+        plys = plys or player.GetAll()
```
Comment (saibotk) [inline lua/ttt2/libraries/entspawnscript.lua:494]:
```suggestion
        plys = plys or playerGetAll()
```

File: lua/ttt2/libraries/entspawnscript.lua [hunk 6]
```diff
@@ -531,7 +531,7 @@ if SERVER then
             end
         end
 
-        plys = plys or select(2, playerIterator())
+        plys = plys or player.GetAll()
```
Comment (saibotk) [inline lua/ttt2/libraries/entspawnscript.lua:534]:
```suggestion
        plys = plys or playerGetAll()
```

**Review by saibotk — CHANGES_REQUESTED** [review]

**Comment by TW1STaL1CKY** [comment]
Sorry to prod, is this fix able to be merged soon? 😄
At the moment the master branch isn't playable without this fix - certain actions like placing a C4 causes the gamemode to completely break

**Comment by TimGoll** [comment]
I'm out of town for this and the next week as well. If someone finishes it, then we can merge it

**Review by saibotk — APPROVED** [review]

**Review by Histalek — APPROVED** [review]

**Pull request merged**